pkg/stanza/fileconsumer/file_windows.go

```diff
@@ -16,6 +17,7 @@ func (m *Manager) preConsume(_ context.Context, _ []*reader.Reader) {
 
 // On windows, we close files immediately after reading because they cannot be moved while open.
 func (m *Manager) postConsume(readers []*reader.Reader) {
-	m.previousPollFiles = readers
+	m.previousPollFiles = m.activeFiles
 	m.closePreviousFiles()
+	m.activeFiles = fileset.New[*reader.Reader](m.maxBatchFiles / 2)
```

Comment:
Same. Also worth nothing that by doing it in `makeReaders`, we aren't managing this separately for different OS.
```suggestion
```